test/browser-pool.test.ts

```diff
@@ -199,6 +199,25 @@ describe('BrowserPool', () => {
             expect(browserPool['_launchBrowser']).toBeCalledTimes(3);
         });
 
+        test('should allow max pages per browser - no race condition', async () => {
+            browserPool.maxOpenPagesPerBrowser = 1;
+            // @ts-expect-error Private function
+            jest.spyOn(browserPool!, '_launchBrowser');
```

Comment:
looks a bit weird that
1. you use `!` here but not elsewhere in the file (so probably not needed)
2. you dont have the expect error comment on the last line of this test, where you also access `_launchBrowser` (so i'd expect either both to fail on non to fail)

Reply:
1. leftover from wip (fixed, thanks)
2. it's accessed through brackets - with the current ts configuration you can access private properties with the brackets operator